resources/migrations/000_migrations.yaml

```diff
@@ -14798,6 +14798,60 @@ databaseChangeLog:
         - customChange:
             class: "metabase.db.custom_migrations.DowngradeDashboardTab"
 
+  - changeSet:
+      id: v47.00-031
+      author: qnkhuat
+      comment: Added 0.47.0 - migrate dashboard grid size from 18 to 24
+      changes:
+          # new_size_x = size_x + ((col + size_x + 1) // 3) - ((col + 1) // 3)
+          # new_col = col + ((col + 1) // 3)
+          # new_size_y = size_y + (size_y + row) // 3 - (row // 3)
+          # new_row = row + (row // 3)
+          - sql:
+              dbms: postgresql,h2
+              sql: >-
+                  update report_dashboardcard set size_x = size_x + floor(cast(size_x + col + 1  as decimal) / 3) - floor(cast(col + 1 as decimal) / 3);
+                  update report_dashboardcard set col    = col + floor(cast(col + 1 as decimal) / 3);
+                  update report_dashboardcard set size_y = size_y + floor(cast(size_y + row as decimal) / 3) - floor(cast(row as decimal) / 3);
+                  update report_dashboardcard set row    = row + floor(cast(row as decimal) / 3);
+          - sql:
+              dbms: mysql,mariadb
+              sql: >-
+                  update report_dashboardcard set size_x = size_x + floor(cast(size_x + col + 1  as decimal) / 3) - floor(cast(col + 1 as decimal) / 3);
+                  update report_dashboardcard set col    = col + floor(cast(col + 1 as decimal) / 3);
+                  update report_dashboardcard set size_y = size_y + floor(cast(size_y + `row` as decimal) / 3) - floor(cast(`row` as decimal) / 3);
+                  update report_dashboardcard set `row`  = `row` + floor(cast(`row` as decimal) / 3);
```

Comment:
Why is `row` necessary? As far as I know it's not a reserved word in mysql or mariadb

Reply:
TIL

Reply:
H2 supports backtick quoting? I normally see it with `"`s